src/MongoDB/Exception/RuntimeException.c

```diff
@@ -23,12 +23,90 @@
 
 #include "phongo_compat.h"
 #include "php_phongo.h"
+#include "php_array_api.h"
 
 zend_class_entry* php_phongo_runtimeexception_ce;
 
+static bool php_phongo_has_string_array_element(zval* labels, char* label TSRMLS_DC)
+{
+	HashTable* ht_data;
+
+	if (Z_TYPE_P(labels) != IS_ARRAY) {
+		return false;
+	}
+
+	ht_data = HASH_OF(labels);
+
+#if PHP_VERSION_ID >= 70000
+	{
+		zval* z_label;
+
+		ZEND_HASH_FOREACH_VAL(ht_data, z_label)
+		{
+			if (Z_TYPE_P(z_label) == IS_STRING) {
+				if (strcmp(Z_STRVAL_P(z_label), label) == 0) {
```

Comment:
I'd be in favor of combining these into a single `if` condition. That won't make the line too long (we have much worse offenders).

Reply:
Done